tests/test_sensor.py

```diff
@@ -80,6 +86,87 @@ def status_func(value):
     assert sensor.status == Sensor.Status.NOMINAL
 
 
+class MyEnum(enum.Enum):
+    ZERO = 0
+    ONE = 1
+
+
+class OtherEnum(enum.Enum):
+    ABC = 0
+    DEF = 1
+
+
+@pytest.mark.parametrize(
+    "sensor_type,compatible_value",
+    [
+        (bool, True),
+        (int, 1_2_3_4),
+        (float, Fraction(1234678)),
+        (float, Fraction(12345678)),
+        (str, "one-two-three-four"),
+        (bytes, b"one-two-three-four"),
+        (Address, Address(IPv4Address("1.2.3.4"))),
+        (Timestamp, 12345678),
+        (MyEnum, MyEnum.ONE),
+    ],
+)
+@pytest.mark.parametrize("initialise_sensor", [True, False])
+def test_sensor_value_setter_success(
+    sensor_type: Type[_T],
+    compatible_value: Any,
+    initialise_sensor: bool,
+) -> None:
+    """Check a compatible value against a sensor type.
+
+    Using `initialise_sensor`, this test checks `compatible_value` either as a
+    default sensor value, or as a new value to an existing sensor.
+    """
+    sensor = Sensor(
+        sensor_type,
+        "test-sensor",
+        default=compatible_value if initialise_sensor else None,
+    )
+    if not initialise_sensor:
+        sensor.value = compatible_value
+    assert sensor.value == compatible_value
+    assert isinstance(sensor.value, sensor_type)
+
+
+@pytest.mark.parametrize(
+    "sensor_type,bad_value",
+    [
+        (bool, "True"),
+        (int, "1234"),
+        (int, 1234.5),
+        (float, "1234.5"),
```

Comment:
Maybe also try `float` with a complex value - that seems like an easy mistake to make.